From b2c03a44866dda105fe8398b3a3aa800f662f37e Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Fri, 8 Sep 2017 13:33:21 +0700
Subject: [PATCH 1/2] fdpm_api_sub: check even for in_width

in_width must be an even number as described in FDPM user manual

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 if/fdpm_api_sub.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/if/fdpm_api_sub.c b/if/fdpm_api_sub.c
index e6f1fd7..d9b0ea4 100755
--- a/if/fdpm_api_sub.c
+++ b/if/fdpm_api_sub.c
@@ -375,7 +375,7 @@ int fdpm_check_seq_par(T_FDP_SEQ *seq_par, int *result_sub_ercd)
 			*result_sub_ercd = E_FDP_PARA_TELECINEMODE;
 			goto exit;
 		}
-		if ((seq_par->in_width < 80) || (seq_par->in_width > 1920)) {
+		if ((seq_par->in_width < 80) || (seq_par->in_width > 1920) || (seq_par->in_width%2)) {
 			ret = -1;
 			*result_sub_ercd = E_FDP_PARA_INWIDTH;
 			goto exit;
-- 
1.9.1

